test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/CreateMockitoMocksCallback.java

```diff
@@ -29,24 +27,25 @@ public void afterConstruct(Object testInstance) {
                 InjectMock injectMockAnnotation = field.getAnnotation(InjectMock.class);
                 if (injectMockAnnotation != null) {
                     boolean returnsDeepMocks = injectMockAnnotation.returnsDeepMocks();
-                    Object contextualReference = getContextualReference(testInstance, field, InjectMock.class);
-                    Optional<Object> result = createMockAndSetTestField(testInstance, field, contextualReference,
+                    InstanceHandle<?> beanHandle = getBeanHandle(testInstance, field, InjectMock.class);
+                    Optional<Object> result = createMockAndSetTestField(testInstance, field, beanHandle,
                             new MockConfiguration(returnsDeepMocks));
                     if (result.isPresent()) {
-                        MockitoMocksTracker.track(testInstance, result.get(), contextualReference);
+                        MockitoMocksTracker.track(testInstance, result.get(), beanHandle.get());
                     }
                 }
             }
             current = current.getSuperclass();
         }
     }
 
-    private Optional<Object> createMockAndSetTestField(Object testInstance, Field field, Object contextualReference,
+    private Optional<Object> createMockAndSetTestField(Object testInstance, Field field, InstanceHandle<?> beanHandle,
             MockConfiguration mockConfiguration) {
-        Class<?> implementationClass = getImplementationClass(contextualReference);
+        Class<?> implementationClass = beanHandle.getBean().getImplementationClass();
         Object mock;
         boolean isNew;
-        Optional<Object> currentMock = MockitoMocksTracker.currentMock(testInstance, contextualReference);
+        // Note that beanHandle.get() returns a client proxy for normal scoped beans; i.e. the contextual instance is not created
```

Comment:
I had to figure this out on my own only to see this comment a few seconds later :-)